src/features/Admin/Rooms/Rooms.tsx

```diff
@@ -1,9 +1,745 @@
-import React from 'react'
+import axios from "axios";
+import React, { useContext, useEffect, useState } from "react";
+import { AuthContext } from "../../../context/AuthContext";
+import {
+  deleteRoomsUrl,
+  faciRoomsUrl,
+  roomsDetailsUrl,
+  roomsUrl,
+  updateRoomsUrl,
+} from "../../../services/api";
+import CustomButton from "../../UI/CustomButton/CustomButton";
+import {
+  AppBar,
+  Button,
+  Checkbox,
+  // CircularProgress,
+  // Dialog,
+  // DialogContent,
+  // DialogTitle,
+  Divider,
+  FormControl,
+  Grid,
+  InputAdornment,
+  InputLabel,
+  ListItemText,
+  Modal,
+  Paper,
+  Select,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableFooter,
+  TableHead,
+  TablePagination,
+  TableRow,
+  TextField,
+  Typography,
+} from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import EditIcon from "@mui/icons-material/Edit";
+import RemoveRedEyeIcon from "@mui/icons-material/RemoveRedEye";
+import { red } from "@mui/material/colors";
+import { Link } from "react-router-dom";
+import styleroom from "./Rooms.module.scss";
+import { IAddRoom } from "../../../interface/RoomInterface";
+import { useForm } from "react-hook-form";
+// import { Box } from "@mui/system";
+import MenuItem from "@mui/material/MenuItem";
+import noData from '../../../assets/images/noData.png'
+import SearchIcon from "@mui/icons-material/Search";
+import noImage from '../../../assets/images/noImage.jpg'
+// import Box from '@mui/material/Box';
+// import Modal from '@mui/material/Modal';
+// import { CloseIcon } from '@mui/icons-material/Close';
 
 const Rooms = () => {
```

Comment:
I am suggesting creating smaller components from this one,  and sharing logic through custom hooks to enhance readability and structure of the application. This will make this component much easier to follow. 

For example, you can make a custom hook  for common data fetching:
useFacilities => to get all facilities and to be used in other parts of the application.
and  in general creating hooks for common logic.
I suggest watching [this video](https://youtu.be/J-g9ZJha8FE?si=Dnaktl6Vg9vPQvOE) for the power of custom hooks

For components, I can see that there are a lot of modals here. You can separate them from here and if needed in any part of the application, you can use them without needing to write the code for them again.